spec/02-integration/03-db/15-connection_pool_spec.lua

```diff
@@ -0,0 +1,110 @@
+local helpers = require "spec.helpers"
+local cjson = require "cjson"
+
+describe("#postgres Postgres connection pool", function()
+  local client
+
+  setup(function()
+    local bp = helpers.get_db_utils("postgres", {
+      "plugins",
+    }, {
+      "slow-query"
+    })
+
+    bp.plugins:insert({
+      name = "slow-query",
+    })
+
+    assert(helpers.start_kong({
+      database = "postgres",
+      nginx_conf = "spec/fixtures/custom_nginx.template",
+      plugins = "slow-query",
+      nginx_worker_processes = 1,
+      pg_pool_size = 1,
+      pg_backlog = 0,
+    }))
+    client = helpers.admin_client()
+  end)
+
+  teardown(function()
+    if client then
+      client:close()
+    end
+    helpers.stop_kong()
+  end)
+
+  it("results in query error too many waiting connect operations", function()
+    local res = assert(client:send {
+      method = "GET",
+      path = "/slow-resource?prime=true",
+      headers = { ["Content-Type"] = "application/json" }
+    })
+    assert.res_status(204 , res)
+
+    res = assert(client:send {
+      method = "GET",
+      path = "/slow-resource",
+      headers = { ["Content-Type"] = "application/json" }
+    })
+    local body = assert.res_status(500 , res)
+    local json = cjson.decode(body)
+    assert.same({ error = "too many waiting connect operations" }, json)
+  end)
+end)
+
+describe("#postgres Postgres connection pool with backlog", function()
+  local client
+
+  setup(function()
+    local bp = helpers.get_db_utils("postgres", {
+      "plugins",
+    }, {
+      "slow-query"
+    })
+
+    bp.plugins:insert({
+      name = "slow-query",
+    })
+
+    assert(helpers.start_kong({
+      database = "postgres",
+      nginx_conf = "spec/fixtures/custom_nginx.template",
+      plugins = "slow-query",
+      nginx_worker_processes = 1,
+      pg_pool_size = 1,
+      pg_backlog = 1,
```

Comment:
```suggestion
      nginx_worker_processes = 1,
      pg_pool_size = 3,
      pg_backlog = 1,
```
This would make the test more meaningful since we are repeating the request 3 times to fill the pool below